api/src/libs/utils.js

```diff
@@ -72,3 +72,39 @@ export const isMatch = (input, target) => {
   }
   return input === target;
 };
+
+/**
+ * Flatten an object with paths for keys
+ * https://bit.ly/2neWfJ2
+ * @param {Object} inputData the object to be flattened
+ * @param {String} prefix prefix for all key path, default as none
+ * @returns {Object} nested object to be flat object with key like 'key1.key2.key3'
+ */
+export const flattenObject = (inputData, prefix = '') =>
+  Object.keys(inputData).reduce((output, k) => {
+    const result = output;
+    // setup key path:
+    const pre = prefix.length ? `${prefix}.` : '';
+    // if the current item is an object (but not an array or null), then loop through:
+    if (typeof inputData[k] === 'object' && !Array.isArray(inputData[k]) && inputData[k])
```

Comment:
could be refactored to lodash's `_.isPlainObject()`

Reply:
actually i think this is a change you'd want to do to make this function more solid.